src/Microsoft.AspNetCore.ResponseCaching/KeyProvider.cs

```diff
@@ -36,9 +36,19 @@ public KeyProvider(ObjectPoolProvider poolProvider, IOptions<ResponseCachingOpti
             _options = options.Value;
         }
 
+        public virtual IEnumerable<string> CreateLookupBaseKey(HttpContext httpContext)
+        {
+            yield return CreateStorageBaseKey(httpContext);
```

Comment:
you may want to check this in the profiler. yield return is easy, but I don't know that it's efficient. Returning a string[1] would probably be more efficient.